sources/packages/backend/apps/queue-consumers/src/processors/schedulers/institution-integration/ier12-integration/ier12-integration.scheduler.ts

```diff
@@ -48,6 +43,7 @@ export class IER12IntegrationScheduler extends BaseScheduler<GeneratedDateQueueI
       job.data.generatedDate,
     );
     this.logger.log("IER 12 file generation completed.");
+    this.schedulerQueue.clean(queueCleanUpPeriod, "completed");
```

Comment:
This can be a protected method on the base class. The base class already has access to `getQueueCleanUpPeriod` and the `schedulerQueue`.
We can have a parameter method called `this.cleanSchedulerQueueHistory()`.